sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp

```diff
@@ -8,7 +8,7 @@
 
 namespace Azure { namespace Storage { namespace Files { namespace DataLake { namespace Details {
   std::string GetBlobUriFromUri(const std::string& uri);
-  std::string GetDfsUriFromUri(const std::string& uri);
+  std::string GetDfsUriFromUri(const std::string& uri, bool returnEmptyOnNotFound = true);
```

Comment:
This is really ugly.

Reply:
There is no better way to do so.

Reply:
Let me ask you one question. When is it useful to return an empty uri if the original uri doesn't contain `.dfs.`?

In [this](https://github.com/Azure/azure-sdk-for-cpp/blob/a898fd819aa0fa987358f751f246f4d8a8b85157/sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp#L62) use case, `m_dfsUri` is empty if the original uri is a customized domain, how is an empty uri better than the original one?

Reply:
You are actually right. I'll remove this.